build_runner/lib/src/asset/reader.dart

```diff
@@ -61,10 +59,13 @@ class SinglePhaseReader implements AssetReader {
 
   @override
   Iterable<AssetId> findAssets(Glob glob) => _assetGraph.allNodes
-      .where((n) => n.id.package == _rootPackage)
-      .where((n) => glob.matches(n.id.path))
-      .where((n) =>
-          n is! GeneratedAssetNode ||
-          (n as GeneratedAssetNode).phaseNumber < _phaseNumber)
-      .map((n) => n.id);
+          .where((n) => n.id.package == _primaryPackage)
+          .where((n) => glob.matches(n.id.path))
+          .where((n) {
+        if (n is GeneratedAssetNode) {
+          return n.wasOutput && n.phaseNumber < _phaseNumber;
```

Comment:
Can you elaborate on the logic change here?

Reply:
This is the other bug fix I mentioned, basically `GeneratedAssetNode`s can represent potential outputs that were never actually output (you can overdeclare outputs).

Previously calls to `findAssets` would give you these non-existent assets and then it would fail if you tried to read them. This fixes that by checking `wasOutput`.

Reply:
Ack. Thanks!